### ISSUE TITLE
[beta] Axios should not be a dependency or peer dependency but a development dependency

### ISSUE THREAD

**Issue by kleinfreund** [Kong/kongponents#817]
Kongponents declares axios both as a dependency and as a peer dependency. It should do neither.

First, it being a peer dependency is unnecessary since axios is already installed as a dependency of the project which means any dependency traversal algorithm will discover the dependency before the peer dependency.

That being said, it should _also_ not be a dependency. Its only code import (besides examples in the documentation) is here:

https://github.com/Kong/kongponents/blob/8a6cff5710430ef416b8195ca27a82d96a36c89a/src/composables/useUtilities.ts#L3

Instead, it should be a development dependency which ensures access to its types during development and while compiling the `useUtilities.ts` file.

This would unfortunately result in a problem because the distribution files of Kongponents do end up having a `dist/types/composables/useUtilities.d.ts` file which includes the same type imports which _would_ create the need for either Kongponents or its consuming project to have axios installed.

I think this shouldn’t be the case. Kuma GUI, for example, doesn’t use axios (anymore), yet when I install or update Kongponents, I get a warning informing me that there is a missing peer dependency (axios).

At the very least, axios shouldn’t be an actual dependency in Kongponents to avoid it being installed in a consuming project which doesn’t use axios (either itself or via Kongponents).

**Comment by adamdehaven** [comment]
> This would unfortunately result in a problem because the distribution files of Kongponents do end up having a dist/types/composables/useUtilities.d.ts file which includes the same type imports which would create the need for either Kongponents or its consuming project to have axios installed.

This is the root of the dilemma as we expect Axios request/response types in our utilization of the `useRequest` composable function in `KTable` and `KCatalog`.

@kleinfreund I put up a potential solution in https://github.com/Kong/kongponents/pull/818 if you want to take a look; however, I don't love it since I'm essentially emulating types from the Axios package that could change out from under us in future releases.

**Comment by kleinfreund** [comment]
@adamdehaven Right, this is unfortunate. I’ve checked earlier and unfortunately, the DefinitelyTyped package for axios is deprecated as axios provides its own types. This would otherwise be ideal.

That being said, these types changing in axios is fairly unlikely and we would still have a good chance to notice it since axios would still be a development dependency.

Alternatively, we could also just move axios to development dependencies without overwriting the types like you did in #818. Would that still compile?

**Comment by adamdehaven** [comment]
> Alternatively, we could also just move axios to development dependencies without overwriting the types like you did in #818. Would that still compile?

It would compile; however, I believe it would break a consuming application as the imports from `axios` would not resolve and would throw a typecheck error

**Comment by kleinfreund** [comment]
Yeah, you’re right. That’s also what I think would happen.

All these things considered, I think maintaining a copy of the types isn’t a terrible compromise considering that it’s unlikely to have frequent breaking changes to them. Alternatively, we could keep axios as a dependency, but that also seems quite heavy-handed. What do you think?

**Comment by adamdehaven** [comment]
I like not requiring Axios; I'm just trying to decide how fragile my solution is. I almost think I need to make my interfaces MORE broad.
